pkg/cloudprovider/provider/gce/types/types.go

```diff
@@ -38,6 +38,8 @@ type CloudProviderSpec struct {
 	Network               providerconfigtypes.ConfigVarString `json:"network"`
 	Subnetwork            providerconfigtypes.ConfigVarString `json:"subnetwork"`
 	Preemptible           providerconfigtypes.ConfigVarBool   `json:"preemptible"`
+	AutomaticRestart      providerconfigtypes.ConfigVarBool   `json:"automaticRestart,omitempty"`
+	ProvisioningModel     providerconfigtypes.ConfigVarString `json:"provisioningModel,omitempty"`
```

Comment:
```suggestion
	AutomaticRestart      *providerconfigtypes.ConfigVarBool   `json:"automaticRestart,omitempty"`
	ProvisioningModel     *providerconfigtypes.ConfigVarString `json:"provisioningModel,omitempty"`
```

These should be optional values